addons/mail/models/mail_message.py

```diff
@@ -959,6 +959,8 @@ def write(self, vals):
     def unlink(self):
         # cascade-delete attachments that are directly attached to the message (should only happen
         # for mail.messages that act as parent for a standalone mail.mail record).
+        if not self:
```

Comment:
What is the purpose of this ?

Reply:
It prevents recursive loop on unlink (according to raphael it should be implemented everywhere compulsorily)